tests/ecosystem/upgrade/test_upgrade_ocp.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import json
+import time
+
+from ocs_ci.ocs import ocp
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.framework.testlib import ManageTest
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.exceptions import CephHealthException
+
+logger = logging.getLogger(__name__)
+
+# TODO: add automatic image finder
+# TODO: add image type validation (ga to ga , nightly to nightly, newer than current etc.)
+# TODO: receive image as external argument
+
+
+class TestUpgradeOCP(ManageTest):
+    """
+    1. check cluster health
+    2. check OCP version
+    3. perform OCP upgrade
+    4. check all OCP ClusterOperators
+    5. check OCP version
+    5. monitor cluster health
+    """
+
+    def test_upgrade_ocp(self):
+        """
+
+        Tests OCS stability when upgrading OCP
+
+        """
+        target_image = "4.3.0-0.nightly-2020-02-17-205936"
+        self.cluster_operators = self.get_all_cluster_operators()
+        ceph_cluster = CephCluster()
+        ceph_cluster.enable_health_monitor()
+
+        logger.info(f" oc version: {self.get_current_oc_version()}")
+
+        self.get_all_cluster_operators()
+
+        # Upgrade OCP
+
+        self.upgrade_ocp(image=target_image)
+
+        # Wait for upgrade
+        for ocp_operator in self.cluster_operators:
+            logger.info(f"Checking upgrade status of {ocp_operator}:")
+            ver = self.get_cluster_operator_version(ocp_operator)
+            logger.info(f"current {ocp_operator} version: {ver}")
+            for attempt in range(1, 100):
```

Comment:
Please use ocs_ci.utility.utils.TimeoutSampler instead of the polling logic here